javatests/arcs/core/entity/HandleManagerTestBase.kt

```diff
@@ -163,106 +165,120 @@ open class HandleManagerTestBase {
+        val handle = writeHandleManager.createSingletonHandle()
 
-        // Verify that clear works on an empty singleton (including the join op).
-        handle.dispatchClear()
-    }
+        // Don't use the dispatchX helpers so we can test the immediate effect of the handle ops.
+        withContext(handle.dispatcher) {
+            // Initial state.
+            assertThat(handle.fetch()).isNull()
 
-    @Test
-    fun singleton_writeAndReadBackAndClear() = testRunner {
-        val writeHandle = writeHandleManager.createSingletonHandle()
-        val readHandle = readHandleManager.createSingletonHandle()
+            // Verify that clear works on an empty singleton.
+            val jobs = mutableListOf<Job>()
+            jobs.add(handle.clear())
+            assertThat(handle.fetch()).isNull()
 
-        var readHandleUpdated = readHandle.onUpdateDeferred()
-        writeHandle.dispatchStore(entity1)
+            // All handle ops should be locally immediate (no joins needed).
+            jobs.add(handle.store(entity1))
+            assertThat(handle.fetch()).isEqualTo(entity1)
+            jobs.add(handle.clear())
+            assertThat(handle.fetch()).isNull()
 
-        // Now read back from a different handle
-        readHandleUpdated.await()
-        val readBack = readHandle.dispatchFetch()
-        assertThat(readBack).isEqualTo(entity1)
-
-        readHandleUpdated = readHandle.onUpdateDeferred()
-        writeHandle.dispatchStore(entity2)
-        readHandleUpdated.await()
-        val readBack2 = readHandle.dispatchFetch()
-        assertThat(readBack2).isEqualTo(entity2)
-
-        readHandleUpdated = readHandle.onUpdateDeferred()
-        writeHandle.dispatchClear()
-        readHandleUpdated.await()
-        val readBack3 = readHandle.dispatchFetch()
-        assertThat(readBack3).isNull()
-    }
-
-    @Test
-    fun singleton_writeAndReadBack() = testRunner {
-        val writeHandle = writeHandleManager.createSingletonHandle()
-        val readHandle = readHandleManager.createSingletonHandle()
-        val readHandleUpdated = readHandle.onUpdateDeferred()
-        writeHandle.dispatchStore(entity1)
-
-        // Now read back from a different handle
-        readHandleUpdated.await()
-        val readBack = readHandle.dispatchFetch()
-        assertThat(readBack).isEqualTo(entity1)
+            // The joins should still work.
+            jobs.joinAll()
+        }
     }
 
     @Test
-    fun singleton_clearOnAClearDataWrittenByA() = testRunner {
-        val handleA = writeHandleManager.createSingletonHandle()
-        val handleB = readHandleManager.createSingletonHandle()
-
-        var handleBUpdated = handleB.onUpdateDeferred()
-        handleA.dispatchStore(entity1)
-        handleBUpdated.await()
-
-        // Now read back from a different handle
-        assertThat(handleB.dispatchFetch()).isEqualTo(entity1)
+    fun singleton_writeAndReadBack_unidirectional() = testRunner {
```

Comment:
instead of 'unidirectional', how about singleWriter (and twoWriters below)?

Reply:
Because it's a write-only sender and a read-only receiver. I don't think "singleWriter" conveys that. Two read-write handles could still have a single writer.